qualtran/bloqs/mod_arithmetic/mod_subtraction.py

```diff
@@ -247,3 +249,158 @@ def _cmod_neg() -> CModNeg:
 
 
 _CMOD_NEG_DOC = BloqDocSpec(bloq_cls=CModNeg, examples=[_cmod_neg])
+
+
+@frozen
+class ModSub(Bloq):
+    r"""Performs modular subtraction.
+
+    Applies the operation $\ket{x} \ket{y} \rightarrow \ket{x} \ket{y-x \% p}$
+
+    Args:
+        dtype: Datatype of the register.
+        p: The modulus for the negation.
+
+    Registers:
+        x: The register contraining the first integer.
+        y: The register contraining the second integer.
+
+    References:
+        [How to compute a 256-bit elliptic curve private key with only 50 million Toffoli gates](https://arxiv.org/abs/2306.08585)
+        Fig 6b and 8
```

Comment:
it's crucial that we get these right. Did you mean 6c? please review carefully here and elsewhere for copy-paste errors

Reply:
fixed ... nice catch